drivers/storage/digitalocean/storage/digitalocean_storage.go

```diff
@@ -0,0 +1,336 @@
+// +build !libstorage_storage_driver libstorage_storage_driver_digitalocean
+
+package storage
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/digitalocean/godo"
+
+	gofig "github.com/akutz/gofig/types"
+	"github.com/akutz/goof"
+	"github.com/codedellemc/libstorage/api/context"
+	"github.com/codedellemc/libstorage/api/registry"
+	"github.com/codedellemc/libstorage/api/types"
+	do "github.com/codedellemc/libstorage/drivers/storage/digitalocean"
+	doUtils "github.com/codedellemc/libstorage/drivers/storage/digitalocean/utils"
+)
+
+type driver struct {
+	name   string
+	config gofig.Config
+	client *godo.Client
+}
+
+func init() {
+	registry.RegisterStorageDriver(do.Name, newDriver)
+}
+
+func newDriver() types.StorageDriver {
+	return &driver{name: do.Name}
+}
+
+func (d *driver) Name() string {
+	return do.Name
+}
+
+func (d *driver) Init(ctx types.Context, config gofig.Config) error {
+	d.config = config
+	token := d.config.GetString(do.ConfigDOToken)
+
+	fields := log.Fields{
+		"token": token,
+	}
+
+	if token == "" {
+		fields["token"] = ""
+	} else {
+		fields["token"] = "******"
+	}
+
+	fields["region"] = d.config.GetString("digitalocean.region")
+
+	client, err := doUtils.Client(token)
+	if err != nil {
+		return err
+	}
+	d.client = client
+
+	log.WithFields(fields).Info("storage driver initialized")
+
+	return nil
+}
+
+func (d *driver) Type(ctx types.Context) (types.StorageType, error) {
+	return types.Block, nil
+}
+
+// DigitalOcean volumes are are found using device-by-id, ex:
+// /dev/disk/by-id/scsi-0DO_Volume_volume-nyc1-01 See
+// https://www.digitalocean.com/community/tutorials/how-to-use-block-storage-on-digitalocean#preparing-volumes-for-use-in-linux
+func (d *driver) NextDeviceInfo(
+	ctx types.Context) (*types.NextDeviceInfo, error) {
+	return nil, nil
```

Comment:
Hi @protochron,

If this is not implemented then please return the [appropriate error](https://github.com/codedellemc/libstorage/blob/master/api/types/types_errors.go#L9), ex. `return nil, types.ErrNotImplemented`.